internal/gatewayapi/securitypolicy.go

```diff
@@ -1672,8 +1672,9 @@ func (t *Translator) buildAPIKeyAuth(
 		namespace: policy.Namespace,
 	}
 
-	credentials := make(map[string]ir.PrivateBytes)
+	credentials := make([]ir.Credential, 0)
```

Comment:
can we prealloc the size of slice?

Reply:
https://sourcegraph.com/github.com/golang/tools/-/blob/gopls/internal/test/marker/testdata/completion/builtins.txt?L79

Have found similar usage here

Reply:
@lalet  I think this is what @zirain meant 👉https://github.com/envoyproxy/gateway/pull/7584#discussion_r2558238375